File: 2021/R/d01.R
part1 <- function(d) {
  sum(d[2:length(d)] > d[1:(length(d) - 1)])
}

part2 <- function(d) {
  sum(d[4:length(d)] > d[1:(length(d) - 3)])
}

d <- as.integer(readLines("../inputs/input_1.txt"))
c(part1(d), part2(d))

File: 2021/PHP/d01.php
<?php
 
  function parse_file($d) {
    $ds = explode("\n", $d);
    foreach ($ds AS $index => $value)
      $ds[$index] = (int) $value; 
    return $ds;
  }

  function solve($d,$lag) {
    $res = 0;
    for ($i = $lag; $i < count($d); $i++)
      $res = $res + ($d[$i] > $d[$i - $lag] ? 1 : 0);
    return $res;
  }

  $d = parse_file(file_get_contents("../inputs/input_1.txt"));
  echo ("Part 1: ".solve($d, 1). "   Part 2: ".solve($d, 3)."\n");

?>